scripts/reana

```diff
@@ -104,6 +106,134 @@ down () {
+                      \"preset_pars\": {\"name\": \"J.Doe\", \"delay\": \"2\"}
+                    }" \
+                    http://"$(reana get reana-workflow-controller)"/yadage \
+                    | grep -oP '(?<="workflow_id": ")[^"]*')
+
+    # Check for successful execution of curl
+    # https://unix.stackexchange.com/a/204790
+    # Note that checking with '$?' might fail on some corner cases
+    # AND when authentication is involved (response codes 401 and 407)
+    submit_status=$?
+
+    if [ "$submit_status" = "0" ]; then
+
+        if [ "$flag_verbose" != "" ]; then
+            echo "[INFO] Open http://$controller_url/$workflow_id to see visual representation of your workflow and it's progress."
+            echo "[INFO] You can 'cat' or 'tail' /reana/$experiment/$workflow_id/yadage/helloworld/greetings.txt -file"
+            echo "       from shared storage of REANA cluster to see how helloworld workflow progresses."
+            echo "[INFO] (e.g. minikube ssh \"tail -c+1 -F /reana/$experiment/$workflow_id/yadage/helloworld/greetings.txt\")"
+        fi
+
+        echo ""
+        echo "Workflow_id: $workflow_id"
+        echo "Monitor: http://$controller_url/$workflow_id"
+        echo "Output:  /reana/$experiment/$workflow_id/yadage/helloworld/greetings.txt"
+        echo ""
+
+        if [ "$flag_devel" != "" ]; then
+            # Start tailing of job-controller logs
+            #kubectl logs -f $(kubectl get pods | grep -Po 'job-controller[^ ]+')
+
+            # Start tailing of yadage-atlas-worker logs
+            #kubectl logs -f $(kubectl get pods | grep -Po 'yadage-atlas-worker[^ ]+')
+
+            # Open workflow-monitor web-interface
+            # Inspired by http://stackoverflow.com/a/27776822
+            os_type=$(uname -s)
+            if [ "$os_type" = "Darwin" ]; then
+                open "http://$controller_url/$workflow_id"
+            elif [ "$os_type" = "Linux" ] ; then
+                xdg-open "http://$controller_url/$workflow_id"
+            fi
+
+            # Tail output-file of the workflow
+            echo "[INFO] Tailing outputfile...hit Ctrl+C when done."
+            echo ""
+            minikube ssh "tail -c+1 -F /reana/$experiment/$workflow_id/yadage/helloworld/greetings.txt"
+        fi
+
+        retval=0
+    else
+        retval=1
+    fi
+
+    return $retval
+}
+
+run () {
+    # Used to start workflows on REANA cluster
+    # Currently hardcoded to run REANA helloworld demo workflow (L210).
+    # (L204) Experiment hardcoded to 'atlas' for now because of
+    # reanahub/reana-workflow-engine-yadage/issues/20
+
+    if [ "$3" = "-e" ]; then
+        experiment=$4
+    else
+        experiment=${REANA_EXPERIMENT-}
+    fi
+
+    workflow=$2
+
+    # The code to check for flags is from http://stackoverflow.com/q/2875424
+    # Not sure if this is a good way (POSIX compliant?) or not.
+    flag_verbose=$(echo "$@" | grep -e "--verbose" -o)
```

Comment:
If I change this to `flag_devel=$(echo $* | grep -e "--devel" -o)`, shellcheck complains:
* SC2048: Use "$@" (with quotes) to prevent whitespace problems.
* SC2086: Double quote to prevent globbing and word splitting.

With `flag_devel=$(echo $@ | grep -e "--devel" -o)` shellcheck complains:
* SC2068: Double quote array expansions, otherwise they're like $* and break on spaces.

If I use `flag_devel=$(echo "$@" | grep -e "--devel" -o)` shellcheck doesn't complain.

I understand your point though.

I don't know enough about shell scripting to be able to say if any of these is good.

Should we change from `sh` to `bash` to use `if [[ "$@" != *"--devel"* ]]; then` if that would be a better alternative?